src/test/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVFTest.java

```diff
@@ -0,0 +1,197 @@
+            .setNbBedrooms(1)
+            .setNbSleeping(4)
+            .setNbBathrooms(1)
+            .setTerrace(true)
+            .setFloorAreaTerrace(40)
+            .setDescription(
+                "Une ferme rustique en compagnie de Dwight Schrute, interdit à Jim Halpert")
+            .setTitle("Une ferme")
+            .setWifi(false)
+            .setPricePerNight(3.3)
+            .setNbMinNight(3)
+            .setTele(false)
+            .build();
+
+    LinearValueFunction floorAreaV = new LinearValueFunction(0d, 200d);
+    builderLinearAVF.setFloorAreaValueFunction(floorAreaV);
+
+    LinearValueFunction nbSleepingV = new LinearValueFunction(3d, 5d);
+    builderLinearAVF.setNbSleepingValueFunction(nbSleepingV);
+
+    ReversedLinearValueFunction nbMinNightV = new ReversedLinearValueFunction(7d, 30d);
+    builderLinearAVF.setNbMinNightValueFunction(nbMinNightV);
+
+    BooleanValueFunction terraceV = new BooleanValueFunction(true);
+    builderLinearAVF.setTerraceValueFunction(terraceV);
+
+    LinearValueFunction nbBedroomsV = new LinearValueFunction(3d, 4d);
+    builderLinearAVF.setNbBedroomsValueFunction(nbBedroomsV);
+
+    ReversedLinearValueFunction pricePerNightV = new ReversedLinearValueFunction(20d, 40d);
+    builderLinearAVF.setPricePerNightValueFunction(pricePerNightV);
+
+    BooleanValueFunction teleV = new BooleanValueFunction(true);
+    builderLinearAVF.setTeleValueFunction(teleV);
+
+    BooleanValueFunction wifiV = new BooleanValueFunction(true);
+    builderLinearAVF.setWifiValueFunction(wifiV);
+
+    LinearValueFunction nbBathroomsV = new LinearValueFunction(2d, 3d);
+    builderLinearAVF.setNbBathroomsValueFunction(nbBathroomsV);
+
+    LinearValueFunction floorAreaTerraceV = new LinearValueFunction(30d, 50d);
+    builderLinearAVF.setFloorAreaTerraceValueFunction(floorAreaTerraceV);
+    for (Criterion c : Criterion.getCriterias()) {
+      builderLinearAVF.setWeightRange(c, Range.closed(5d, 15d));
+    }
+    linearAVF = builderLinearAVF.build();
+  }
+
+  /** Function to test the some setters */
+  @Test
+  void checkValue() {
+    LinearValueFunction lvf = linearAVF.getNbSleepingValueFunction();
+    assertEquals(5d, lvf.getInterval().upperEndpoint());
+    lvf = linearAVF.getNbBedroomsValueFunction();
+    assertEquals(4d, lvf.getInterval().upperEndpoint());
+    lvf = linearAVF.getNbBathroomsValueFunction();
+    assertEquals(3d, lvf.getInterval().upperEndpoint());
+  }
+
+  /** Function to test the computing of the subjective value of an apartment */
+  @Test
+  void linearAVFTest() {
+    assertEquals(0.5, linearAVF.getSubjectiveValue(a), 0.0001);
+    linearAVF.setWeightRange(Criterion.TELE, Range.closed(2d, 8d));
+    assertEquals(10d, linearAVF.getMiddleOfRange(Criterion.TELE));
+    assertEquals(0.5, linearAVF.getSubjectiveValue(a), 0.00001);
+  }
+
+  /** Test if the weight setter throw a Illegal Argument Exception when needed */
+  @Test
+  void exceptionIllegalArgWeightSetter() {
```

Comment:
Évitez les tests qui ne font que répéter votre code. Ceci est inutile.

Reply:
fixed in c76efc5fdf078f40c9822c59c63e72464723fe50